protocol/river-options-unstable-v1.xml

```diff
@@ -0,0 +1,143 @@
+      <description summary="get an option handle for the given key">
+        If the output argument is non-null, the option is local to the given
+        output. Otherwise it is considered global.
+      </description>
+      <arg name="key" type="string"/>
+      <arg name="output" type="object" interface="wl_output" allow-null="true"/>
+      <arg name="handle" type="new_id" interface="zriver_option_handle_v1"/>
+    </request>
+  </interface>
+
+  <interface name="zriver_option_handle_v1" version="1">
+    <description summary="handle to an option">
+      On binding this object, one of the events will immediately be sent by
+      the server to inform the client of the current state of the option. New
+      events will be sent as the state changes.
+    </description>
+
+    <request name="destroy" type="destructor">
+      <description summary="destroy the handle">
+        This request indicates that the client will not use the
+        zriver_option_handle_v1 any more and that it may be safely destroyed.
+      </description>
+    </request>
+
+    <event name="unset">
+      <description summary="the option is currently unset">
+        The option with this key has never been set, so the first set_*_value
+        request received from any client will determine its type.
+
+        This can only ever be sent as the first event after binding this
+        interface as options cannot be unset once set.
+      </description>
+    </event>
+
+    <event name="int_value">
+      <description summary="the current value of the int option">
+        This indicates to the client that the option is of type int as well
+        as the current value of the option.
+      </description>
+      <arg name="value" type="int"/>
+    </event>
+
+    <event name="uint_value">
+      <description summary="the current value of the uint option">
+        This indicates to the client that the option is of type uint as well
+        as the current value of the option.
+      </description>
+      <arg name="value" type="uint"/>
+    </event>
+
+    <event name="fixed_value">
+      <description summary="the current value of the fixed option">
+        This indicates to the client that the option is of type fixed as
+        well as the current value of the option.
+      </description>
+      <arg name="value" type="fixed"/>
+    </event>
+
+    <event name="string_value">
+      <description summary="the current value of the string option">
+        This indicates to the client that the option is of type string as
+        well as the current value of the option.
+      </description>
+      <arg name="value" type="string" allow-null="true"/>
+    </event>
+
+    <request name="set_int_value">
+      <description summary="set the value of the option">
+        If the option is either unset or set to a value of type int,
+        this request sets the value of the option as well as the type if
+        previously unset.
```

Comment:
This does not mention what happens when the type is not the same as that of the request. Maybe we should have an additional sentence like "Otherwise is is ignored."

Reply:
It should definitely not cause a protocol error. That way we keep it simple for clients.

Reply:
Yes, the intent is that the request is simply ignored if the type does not match. I agree it should be stated more explicitly though.